### PR TITLE
PYTHON-4187 Ensure secrets are not logged in Evergreen

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -380,6 +380,7 @@ functions:
       params:
         working_dir: "src"
         shell: bash
+        include_expansions_in_env: true
```
Comment (ShaneHarvey) [inline .evergreen/config.yml:383]: IIUC this should be a list of vars to avoid pulling in vars we don't need.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: .evergreen/config.yml (hunk 1 above)
Comment (blink1073) [reply .evergreen/config.yml:383]: According to the mitigation document, this is an appropriate solution for now.

Comment (blink1073) [reply .evergreen/config.yml:383]: Updating to use add_expansions_to_env, which we have used in several places.  I'll open up a follow-up ticket to make them all explicit.

Comment (ShaneHarvey) [reply .evergreen/config.yml]: Wasn't there an issue where add_expansions_to_env caused AWS env vars to be injected thus causing issues in the MONGODB-AWS tests? If not I'm fine with your approach.

**Review by Jibola — APPROVED** [review]
LGTM

**Pull request merged**